src/core/common/tracker.hpp

```diff
@@ -0,0 +1,355 @@
+/*
+ *  Copyright (c) 2017, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definitions for tracker class for OpenThread objects.
+ */
+
+#ifndef TRACKER_HPP_
+#define TRACKER_HPP_
+
+#include <openthread/config.h>
+
+#include <openthread/types.h>
+
+#include "openthread-core-config.h"
+#include "openthread-single-instance.h"
+
+namespace ot {
+
+class ThreadNetif;
+class MeshForwarder;
+class TaskletScheduler;
+class TimerScheduler;
+namespace Ip6 { class Ip6; }
+
+/**
+ * @addtogroup core-tracker
+ *
+ * @brief
+ *   This module includes definitions for tracker base class for OpenThread objects.
+ *
+ * @{
+ *
+ */
+
+/**
+ * This template class implements the base tracker for OpenThread objects.
+ *
+ */
+template <typename TrackedType>
+class Tracker
```

Comment:
Yes. Thanks. I like the "Locator" option (e.g., `ThreadNetifLocator` `InstanceLocator`...

(I went back and forth between "Tracker" and "Keeper"...also considered "Maintainer", "Saver", and simply calling it "Base"... :)...)